src/mps_utils.jl

```diff
@@ -55,10 +55,26 @@ function parse_protocol(ahs_program, τ::Float64, n_τ_steps::Int)
 
     time_points_Ω = ahs_program["hamiltonian"]["drivingFields"][1]["amplitude"]["time_series"]["times"]
     values_Ω = ahs_program["hamiltonian"]["drivingFields"][1]["amplitude"]["time_series"]["values"]
-
-    pattern = ahs_program["hamiltonian"]["shiftingFields"][1]["magnitude"]["pattern"]
-    time_points_shift = ahs_program["hamiltonian"]["shiftingFields"][1]["magnitude"]["time_series"]["times"]
-    values_shift = ahs_program["hamiltonian"]["shiftingFields"][1]["magnitude"]["time_series"]["values"]
+        
+    n_atoms = length(ahs_program["setup"]["ahs_register"]["sites"])
+    # Supporting both localDetuning and shiftingFields naming conventions
+    loc_detuning_key = ""
+    if "shiftingFields" ∈ keys(ahs_program["hamiltonian"])
+        loc_detuning_key = "shiftingFields"
+    elseif "localDetuning" ∈ keys(ahs_program["hamiltonian"])
+        loc_detuning_key = "localDetuning"
+    end
+    
+    # Check if local detuning or shifting fields were found, otherwise assign zeros
+    if !isempty(loc_detuning_key)
```

Comment:
I think you can simplify this with [`Base.get`](https://docs.julialang.org/en/v1/base/collections/#Base.get)